pkg/destroy/vsphere/vsphere.go

```diff
@@ -189,11 +189,19 @@ func deleteTag(ctx context.Context, client *rest.Client, tagID string) error {
 
 func deleteTagCategory(ctx context.Context, client *rest.Client, categoryID string) error {
 	tagManager := tags.NewManager(client)
-	category, err := tagManager.GetCategory(ctx, categoryID)
-	if err == nil {
-		err = tagManager.DeleteCategory(ctx, category)
+	ids, err := tagManager.ListCategories(ctx)
+	if err != nil {
+		return err
 	}
-	return err
+
+	for _, id := range ids {
+		category, err := tagManager.GetCategory(ctx, id)
+		if err == nil && category.Name == categoryID {
```

Comment:
We should not be ignoring all errors here. We can ignore not found errors, so that we don't get into the same problem that the `GetCategories` function has in govmomi. But if an error is something else, such as failure to connect, then that error should be handled.